### DISCUSSION TITLE
No supported websocket library

### DISCUSSION THREAD

**Discussion by csrgxtu** [encode/starlette#1860] [Q&A]
```
WARNING:  Unsupported upgrade request.
WARNING:  No supported WebSocket library detected. Please use 'pip install uvicorn[standard]', or install 'websockets' or 'wsproto' manually.
```
![image](https://user-images.githubusercontent.com/5053620/190571121-ad4dc47c-8de0-4fd2-868b-8ac9facc5554.png)

what if other languages websocket client, like javascript, will starlette support it?

**Comment by euri10** [comment]
this is a server message from uvicorn, it means the server does not have a websocket library it can use to process incoming requests. so it wont be able to deal with ws requests and tells you to install one of the 2 websicket library uvicorn is using.

however once the server can process ws, you can use whatever websocket client language you want
